packages/babel-helpers/src/helpers.js

```diff
@@ -413,6 +416,29 @@ helpers.interopRequireWildcard = template(`
   })
 `);
 
+helpers.specRequireInterop = template(`
+  (function (obj) {
+    if (obj && obj.__esModule) {
+      return obj;
+    } else {
+      var newObj = Object.create(null, {
+          default: {
+            value: obj,
+            writable: true,
+            enumerable: true
+          },
+          __esModule: {
```

Comment:
The `__esModule` key is already not writable/enumerable/configurable in the regular transform; it's only enumerable with `loose: true`. I guess `Object.isFrozen` would be the only reliable way without needing `Symbol.toStringTag`